crypto/rsa_extra/rsa_test.cc

```diff
@@ -839,52 +867,80 @@ TEST(RSATest, CheckKey) {
   bssl::UniquePtr<RSA> rsa(RSA_new());
   ASSERT_TRUE(rsa);
 
+  bssl::UniquePtr<EVP_PKEY> rsa_pkey(EVP_PKEY_new());
+  ASSERT_TRUE(rsa_pkey);
+  ASSERT_TRUE(EVP_PKEY_set1_RSA(rsa_pkey.get(), rsa.get()));
+
   // Missing n or e does not pass.
   ASSERT_TRUE(BN_hex2bn(&rsa->n, kN));
   EXPECT_FALSE(RSA_check_key(rsa.get()));
+
+  bssl::UniquePtr<EVP_PKEY_CTX> rsa_key_ctx(
+          EVP_PKEY_CTX_new(rsa_pkey.get(), NULL));
+  ASSERT_TRUE(rsa_key_ctx);
+  EXPECT_FALSE(EVP_PKEY_check(rsa_key_ctx.get()));
+  EXPECT_FALSE(EVP_PKEY_public_check((rsa_key_ctx.get())));
```

Comment:
Is this test separate than what's on 875? Should the clear error be moved up there?